src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -16,46 +16,76 @@
 
 using System.Diagnostics.Metrics;
 using System.Reflection;
+using System.Threading;
 using Diagnostics = System.Diagnostics;
 
 namespace OpenTelemetry.Instrumentation.Process;
 
-internal class ProcessMetrics
+internal sealed class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
 
-    static ProcessMetrics()
+    private static readonly ThreadLocal<InstrumentsValues> CurrentThreadInstrumentsValues = new(() => new InstrumentsValues());
+
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.usage",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
-            },
+            () => CurrentThreadInstrumentsValues.Value.GetMemoryUsage(),
             unit: "By",
             description: "The amount of physical memory in use.");
 
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.virtual",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
-            },
+            () => CurrentThreadInstrumentsValues.Value.GetVirtualMemoryUsage(),
             unit: "By",
             description: "The amount of committed virtual memory.");
     }
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ProcessMetrics"/> class.
-    /// </summary>
-    /// <param name="options">The options to define the metrics.</param>
-    public ProcessMetrics(ProcessInstrumentationOptions options)
+    private sealed class InstrumentsValues
     {
+        private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
+        private double? memoryUsage;
+        private double? virtualMemoryUsage;
+
+        internal InstrumentsValues()
+        {
+            this.memoryUsage = null;
+            this.virtualMemoryUsage = null;
+        }
+
+        internal double GetMemoryUsage()
+        {
+            if (!this.memoryUsage.HasValue)
+            {
+                this.Snapshot();
+            }
+
+            var value = (double)this.memoryUsage;
+            this.memoryUsage = null;
+            return value;
+        }
+
+        internal double GetVirtualMemoryUsage()
```

Comment:
What You think instead of having two methods `GetMemoryUsage()` and `GetVirtualMemoryUsage()`, you can have only one method returning both values as value tuple? something like the following:

```C#
public static (double memoryUsage, double virtualMmeoryUsage) GetProcessMeasurements() => (CurrentProcess.WorkingSet64, CurrentProcess.VirtualMemorySize64);
```

and you can get rid of the whole class and the static local variable. Or do you think users will need to get each individual value at a different time?

Reply:
My understanding is that the flexibility of this level of customization is deliberately provided by OTel SDK with the use of [View](https://github.com/open-telemetry/opentelemetry-specification/blob/31c81e87d697d5fc6dcff29eee24d6a0b0dbba26/specification/metrics/sdk.md#view).

But this grouping idea could potentially be used on instruments that would not make much sense if being decoupled. A good case for ProcessMetrics is actually process physical memory usage and process virtual memory usage. 

I am still debating myself whether to add this "limitation"(?) to ProcessMetrics. And if added, what should be the behavior of ProcessMetrics when there are users who only care about process virtual memory usage.